src/operator/effectivepolicy/Syncer.go

```diff
@@ -0,0 +1,142 @@
+package effectivepolicy
+
+import (
+	"context"
+	goerrors "errors"
+	"github.com/amit7itz/goset"
+	"github.com/otterize/intents-operator/src/operator/api/v1alpha3"
+	"github.com/otterize/intents-operator/src/shared/errors"
+	"github.com/otterize/intents-operator/src/shared/injectablerecorder"
+	"github.com/otterize/intents-operator/src/shared/serviceidresolver/serviceidentity"
+	"github.com/samber/lo"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type EffectivePoliciesApplier interface {
+	ApplyEffectivePolicies(ctx context.Context, eps []ServiceEffectivePolicy) (int, error)
+	InjectRecorder(recorder record.EventRecorder)
+}
+
+type Syncer struct {
```

Comment:
I would like to have a name that better communicates what this does. I get that it Syncs, but that's a rather generic word and this does a very specific thing.

ClusterwideEffectivePolicySyncer?

Or maybe (thinking out loud), let's think about what it does:
1. Calculates effective policies for all services in the cluster
2. Triggers "apply" (or reconcile if we take my other comment into account)

So perhaps it could be ReconcileAll? If the EffectivePoliciesAppliers turn into EffectivePoliciesReconcilers.